docker/configuration/supervisord.conf

```diff
@@ -0,0 +1,20 @@
+# supervisord.conf
+
+[supervisord]
+nodaemon=true
+
```

Comment:
do we need to add these here?
```
logfile=/dev/null
logfile_maxbytes=0
```
here, as suggested [here](https://docs.docker.com/engine/containers/multi-service_container/#use-a-process-manager) (there is an example at the bottom). I believe these is to prevent accumulating supervisord's own logs on container.

Reply:
Good point. We could configure the log file maxbytes and the number of backups kept for all programs (supervisord, caddy, and python_app). Something as below.

```
[supervisord]
nodaemon=true
logfile=/dev/stdout
logfile_maxbytes=5MB
logfile_backups=3

[program:caddy]
command=caddy run --config /etc/caddy/Caddyfile --adapter caddyfile
autorestart=true
stdout_logfile=/dev/stdout
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=3
stderr_logfile=/dev/stderr
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=3

[program:python_app]
command=/home/annotator/venv/bin/python -m biothings_annotator --conf=/home/annotator/configuration/default.json
autorestart=true
stdout_logfile=/dev/stdout
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=3
stderr_logfile=/dev/stderr
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=3

```

I'm not sure about the values for maxbytes and backups yet.
 
If we use `logfile=/dev/null` the logs are going to be discarded.

Reply:
Yes, however you think it's proper there. The main purpose is to make sure the container itself does not accumulate local files over the time.

Reply:
Looking in the `supervisor` documentation. we have this:

- `logfile`: If logfile is set to a special file like /dev/stdout that is not seekable, log rotation must be disabled by setting logfile_maxbytes = 0.

- `logfile_backups`: The number of backups to keep around resulting from activity log file rotation. If set to 0, no backups will be kept

- `stdout_logfile`: If stdout_logfile is set to a special file like /dev/stdout that is not seekable, log rotation must be disabled by setting stdout_logfile_maxbytes = 0

- `stdout_logfile_backups`: The number of stdout_logfile backups to keep around resulting from process stdout log file rotation. If set to 0, no backups will be kept.

- `stderr_logfile`: If stderr_logfile is set to a special file like /dev/stderr that is not seekable, log rotation must be disabled by setting stderr_logfile_maxbytes = 0.

- `stderr_logfile_backups`: The number of backups to keep around resulting from process stderr log file rotation. If set to 0, no backups will be kept.


Considering the documentation, the attributes `maxbytes` and `backups`could be set to `0`. In this case, the logs are directed to /dev/stdout and /dev/stderr: the configuration sends logs from `caddy` and `python_app` to `stdout` and `stderr`, which are captured by Docker and not stored as files inside the container. Docker then streams these logs to the configured AWS CloudWatch Logs service. Because `stdout_logfile` and `stderr_logfile` are set to /dev/stdout and /dev/stderr, `Supervisor` is not writing logs to the container’s file system. The container itself will not accumulate log files, avoiding any disk space issues.

### Reference
- http://supervisord.org/configuration.html#supervisord-section-values